sljit_src/sljitExecAllocator.c

```diff
@@ -168,17 +168,45 @@ static SLJIT_INLINE void apple_update_wx_flags(sljit_s32 enable_exec)
 #define SLJIT_MAP_JIT	(0)
 #endif /* !SLJIT_MAP_JIT */
 
+#ifdef __FreeBSD__
+#include <sys/procctl.h>
+#include <limits.h>
+#endif
+
+static SLJIT_INLINE void *sljit_map_wx(size_t size, int flags, int fd)
+{
+	void *ptr;
+	int prot;
+       
+	prot = PROT_READ | PROT_WRITE | PROT_EXEC;
+#ifdef PROT_MAX
+	prot |= PROT_MAX(prot);
+#endif
+
+	ptr = mmap(NULL, size, prot, flags, fd, 0);
+	if (ptr != MAP_FAILED)
+		return ptr;
+#ifdef PROC_WXMAP_CTL
+	static int r = INT_MIN;
+	if (r == INT_MIN) {
+		int sljit_wx_enable = PROC_WX_MAPPINGS_PERMIT;
+		r = procctl(P_PID, 0, PROC_WXMAP_CTL, &sljit_wx_enable);
```

Comment:
checking for the right kind of error from this function would be better, since AS-IS it would just trigger EINVAL and disable JIT in systems that don't understand PROX_WXMAP_CTL (a 13.1 binary running in older FreeBSD versions)